src/main/java/ch/njol/skript/CustomEvents.java

```diff
@@ -0,0 +1,148 @@
+				if (entry != null) {
+					File destination = new File(Skript.getInstance().getDataFolder(), entry.getName());
+					if (!destination.exists())
+						saveTo = destination;
+				} if (saveTo != null) {
+					try (InputStream in = zip.getInputStream(entry)) {
+						FileUtils.save(in, saveTo);
+					}
+				}
+			} catch (IOException e) {
+				if (Skript.debug())
+					Skript.exception(e);
+			} finally {
+				eventsFile = new File(Skript.getInstance().getDataFolder(), "custom-events.sk");
+				try {
+					configuration = new Config(eventsFile, false, false, ":");
+				} catch (IOException e) {
+					e.printStackTrace();
+				}
+			}
+		}
+
+		if (configuration != null) {
+			configuration.load(CustomEvents.class);
+			// When we need to change values in the future of this file, bump the version node.
+			if (configuration.get("version").isEmpty() || !configuration.get("version").equalsIgnoreCase("1")) {
+				Skript.warning("Your custom-events.sk config file is outdated. " +
+						"Backup any changes you've made, and delete your custom-events.sk in the Skript folder to update it. " +
+						"After, re-add any nodes you've changed.");
+			}
+			SectionNode section = (SectionNode) configuration.getMainNode().get("events");
+			if (section != null && !section.isEmpty()) {
+				for (Node node : section) {
+					if (!(node instanceof SectionNode)) {
+						Skript.warning("Event node '" + node.getKey() + "' was not a section, ignoring.");
+						continue;
+					}
+					SectionNode eventSection = (SectionNode) node;
+					String name = node.getKey();
+					String enabled = eventSection.get("enabled", "true");
+					if (enabled.equalsIgnoreCase("false"))
+						continue;
+					String description = eventSection.get("description", "custom event '" + name + "'");
+					String pattern = eventSection.getValue("pattern");
+					if (pattern == null) {
+						Skript.warning("Event node '" + name + "' did not contain a 'pattern' value.");
+						continue;
+					}
+					if (pattern.contains("%")) {
+						SkriptLogger.setNode(eventSection.get("pattern"));
+						Skript.warning("Event node '" + name + "' cannot contain any expressions in the pattern '" + pattern + "'");
+						continue;
+					}
+					String eventClass = eventSection.getValue("class");
+					if (eventClass == null) {
+						Skript.warning("Event node '" + name + "' did not contain a 'class' value.");
+						continue;
+					}
+
+					// Everything below this should be dedicated to the class node.
+					SkriptLogger.setNode(eventSection.get("class"));
+					Class<?> event;
+					try {
+						event = Class.forName(eventClass);
+					} catch (Exception e) {
+						if (Skript.logVeryHigh() || Skript.debug()) {
+							Skript.exception(e, "Class '" + eventClass + "' for event node '" + name + "' had an exception loading.");
+						} else {
+							Skript.error("Failed to find class '" + eventClass + "' for event node '" + name + "'");
+						}
+						continue;
+					}
+					if (event == null)
```

Comment:
event should not be able to be null here, since Class.forName is annotated as NotNull